core/src/Revolution/Formatter/modManagerDateFormatter.php

```diff
@@ -0,0 +1,246 @@
+        $format = $this->managerDateFormat . $managerDateTimeSeparator . $this->managerTimeFormat;
+
+        return $this->format($value, $format, $useOffset, $emptyValue);
+    }
+
+    /**
+     * Formats a Resource-related date when applicable and includes localized default text to
+     * represent conditions where a date is not present or applicable
+     *
+     * @param string|int $value The source date value to format
+     * @param string $whichDate An identifier specifying the type of date being formatted
+     * @param bool $useStandardEmptyValue Whether to use the default empty value defined in the system settings
+     * @return string The formatted date or relevant text indicating an empty date value
+     */
+    public function formatResourceDate($value, string $whichDate = 'created', bool $useStandardEmptyValue = true): string
+    {
+        if ($useStandardEmptyValue) {
+            $emptyValue = null;
+        } else {
+            switch ($whichDate) {
+                case 'edited':
+                    $emptyValue = $this->modx->lexicon('unedited');
+                    break;
+                case 'publish':
+                case 'unpublish':
+                    $emptyValue = $this->modx->lexicon('notset');
+                    break;
+                case 'published':
+                    $emptyValue = $this->modx->lexicon('unpublished');
+                    break;
+                default:
+                    $emptyValue = $this->modx->lexicon('unknown');
+            }
+            $emptyValue = '(' . $emptyValue . ')';
+        }
+
+        return $this->formatDateTime($value, true, $emptyValue);
+    }
+
+    /**
+     * Formats a Package-related date when applicable and includes localized default text to
+     * represent conditions where a date is not present or applicable
+     *
+     * @param string|int $value The source date value to format
+     * @param string $whichDate An identifier specifying the type of date being formatted
+     * @param bool $useStandardEmptyValue Whether to use the default empty value defined in the system settings
+     * @return string The formatted date or relevant text indicating an empty date value
+     */
+    public function formatPackageDate($value, string $whichDate = 'created', bool $useStandardEmptyValue = true): string
+    {
+        if ($useStandardEmptyValue) {
+            $emptyValue = null;
+        } else {
+            switch ($whichDate) {
+                case 'installed':
+                    $emptyValue = $this->modx->lexicon('not_installed');
+                    break;
+                case 'updated':
+                    $emptyValue = $this->modx->lexicon('not_updated');
+                    break;
+                default:
+                    $emptyValue = $this->modx->lexicon('unknown');
+            }
+            $emptyValue = '(' . $emptyValue . ')';
+        }
+
+        return $this->formatDateTime($value, true, $emptyValue);
```

Comment:
I think we can lower the complexity of this fn, by doing:
```suggestion
        if ($useStandardEmptyValue) {
            return $this->formatDateTime($value, true, null);
        }

        switch ($whichDate) {
            case 'installed':
                $emptyValue = $this->modx->lexicon('not_installed');
                break;
            case 'updated':
                $emptyValue = $this->modx->lexicon('not_updated');
                break;
            default:
                $emptyValue = $this->modx->lexicon('unknown');
        }
        $emptyValue = '(' . $emptyValue . ')';

        return $this->formatDateTime($value, true, $emptyValue);
```

Reply:
Done